### DISCUSSION TITLE
It is not possible to add any extension through the code

### DISCUSSION THREAD

**Discussion by lucasmarquex** [daijro/camoufox#46] [Q&A]
It is not possible to add any extension through the code, whether xpi or folder. Extracts the files from capsolver, but it does not add them to the browser

**Comment by daijro** [comment]
Hello, could you provide a snippet of your code?

Make sure the path you entered into `addons` contains a manifest.json file.

```python
from camoufox.sync_api import Camoufox

with Camoufox(addons=['/path/to/addon']) as browser:
    page = browser.new_page()
```

http://camoufox.com/fingerprint/addons

**Comment by lucasmarquex** [comment]
`import time
from camoufox.sync_api import Camoufox, Screen
from camoufox import DefaultAddons

with Camoufox(
    geoip=True,
    proxy={
        'server': 'http://192.168.1.220:3128',
        'username': '',
        'password': ''
    },
    os=('windows'),
    screen=Screen(max_width=1920, max_height=1080),
    humanize=True,
    headless=False,
    addons=['capsolver']
) as browser:`

**Comment by lucasmarquex** [comment]
capsolver and the folder with all the files unzipped from the link's zip

**Reply by daijro** [reply]
Could you try using the full path to the capsolver folder?

**Reply by lucasmarquex** [reply]
I tried the full path, from C:, it didn't work
